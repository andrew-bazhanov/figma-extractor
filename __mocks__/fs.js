import { fs } from 'memfs';

fs.__originalFs = jest.requireActual('fs');
module.exports = fs;
