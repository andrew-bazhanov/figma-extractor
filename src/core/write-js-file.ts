import fs from 'fs';
import path from 'path';

const fileLengthRuleDisableComment = '/* eslint-disable max-lines */';

const namingConventionRuleDisableComment =
  '/* eslint-disable @typescript-eslint/naming-convention */';

// NOTE: temporarily changed writeFile to writeFileSync to make testing easier:
// it is unclear how to test callback version without introducing promises.
// But even with promises, the logic seems to "just write" contents without returning the promise itself,
// which is confusing a bit, because it is impossible to handle promise fulfillment.
export const writeJsFile = (template: string, filePath: string) => {
  const dir = path.dirname(filePath);

  // Create directory in advance if there is no one
  if (dir && !fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }

  try {
    fs.writeFileSync(
      filePath,
      `
      ${fileLengthRuleDisableComment}
      ${namingConventionRuleDisableComment}
      ${template}
      `,
    );
    console.log(`Generate ${filePath} is completed`);
  } catch {
    console.log(`Formatting ${filePath} file...`);
  }
};
