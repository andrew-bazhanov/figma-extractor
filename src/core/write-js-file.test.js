import fs from 'fs';
import { vol } from 'memfs';
import path from 'path';

import { writeJsFile } from './write-js-file';

jest.mock('fs');

describe('writeJsFile', () => {
  afterEach(() => {
    vol.reset();
  });

  it('checks existing of a new file', () => {
    expect(fs.existsSync('/ui/theme')).toBe(false);

    writeJsFile(
      `
    module.exports = {
      a: 1,
      b: 2,
      c: 3,
    };
    `,
      path.join('/ui/theme/', 'file-name.js'),
    );

    expect(fs.existsSync('/ui/theme')).toBe(true);
  });

  it('writes test data into a new file', () => {
    writeJsFile(
      `
    module.exports = {
      a: 1,
      b: 2,
      c: 3,
    };
    `,
      path.join('/ui/theme/', 'file-name.js'),
    );

    expect(vol.toJSON()).toMatchSnapshot();
  });
});
