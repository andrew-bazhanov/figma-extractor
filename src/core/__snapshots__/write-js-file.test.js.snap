// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeJsFile writes test data into a new file 1`] = `
{
  "/ui/theme/file-name.js": "
      /* eslint-disable max-lines */
      /* eslint-disable @typescript-eslint/naming-convention */
      
    module.exports = {
      a: 1,
      b: 2,
      c: 3,
    };
    
      ",
}
`;
